src/Service/PagePurger.php

```diff
@@ -30,16 +31,88 @@ public function __construct( MediawikiApi $api ) {
 	/**
 	 * @since 0.3
 	 *
-	 * @param Page $page
+	 * @brief Purge a single page
 	 *
-	 * @return bool
+	 * Purges a single page by submitting a
+	 * 'purge' action to the mediawiki api
+	 * with the parameter 'pageids' set to
+	 * the singe page id
+	 *
+	 * @param Page $page the page that is going to be purged
+	 *
+	 * @return bool return true if the purge was successful
 	 */
 	public function purge( Page $page ) {
-		$this->api->postRequest(
+		$responseArray = $this->api->postRequest(
 			new SimpleRequest( 'purge', [ 'pageids' => $page->getId() ] )
 		);
 
-		return true;
+		// the purge response for the page
+		$purgeResponse = $responseArray['purge'][0];
```

Comment:
This and the other occurance at line 95 make tests to fail, is it a good idea to place the following workaround ?

```php
// if the purge action failed unexpected
if ( !array_key_exists( 'purge' , $responseArray) ) {
   return false; / return new Pages();
}
```